File: src/main/java/com/patrickgrimard/MessageServiceImpl.java
package com.patrickgrimard;

import org.springframework.cloud.context.config.annotation.RefreshScope;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Service;

import javax.inject.Inject;

/**
 * Created on 2016-03-01
 *
 * @author Patrick
 */
@Service
@RefreshScope
public class MessageServiceImpl implements MessageService {

    private Environment env;

    @Inject
    public MessageServiceImpl(Environment env) {
        this.env = env;
    }

    public String getMessage() {
        return env.getProperty("message");
    }
}

File: src/main/kotlin/com/patrickgrimard/MessageServiceImpl.kt
package com.patrickgrimard

/**
 *
 *
 * Created on 2016-03-24
 *
 * @author Patrick
 */
@org.springframework.stereotype.Service
open class MessageServiceImpl : MessageService {
    override fun getMessages(): Array<Message> {
        return arrayOf(Message(0, "Hello"), Message(1, "Spring Rocks!"), Message(2, "Thought you would like to know..."))
    }
}